trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/TestOpaSecurity.java

```diff
@@ -0,0 +1,150 @@
+import static io.trino.aws.proxy.server.security.opa.OpaS3SecurityModule.OPA_S3_SECURITY_IDENTIFIER;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@TrinoAwsProxyTest(filters = TestOpaSecurity.Filter.class)
+public class TestOpaSecurity
+{
+    private final HttpClient httpClient;
+    private final int containerPort;
+    private final S3Client s3Client;
+
+    public static class Filter
+            implements BuilderFilter
+    {
+        @Override
+        public TestingTrinoAwsProxyServer.Builder filter(TestingTrinoAwsProxyServer.Builder builder)
+        {
+            return builder.withProperty("s3-security.type", OPA_S3_SECURITY_IDENTIFIER)
+                    .withProperty("opa-s3-security.server-base-uri", "http://localhost")
+                    .withProperty("opa-s3-security.data-path", "/v1/data/test")
+                    .addModule(binder -> binder.bind(OpaS3SecurityMapper.class).to(OpaMapper.class).in(Scopes.SINGLETON))
+                    .withS3Container()
+                    .withOpaContainer();
+        }
+    }
+
+    public static class OpaMapper
+            implements OpaS3SecurityMapper
+    {
+        private final int containerPort;
+
+        @Inject
+        public OpaMapper(OpaContainer container)
+        {
+            containerPort = container.getPort();
+        }
+
+        @Override
+        public Optional<String> tableName(ParsedS3Request request, Optional<String> lowercaseAction)
+        {
+            return Optional.of(request.keyInBucket());
+        }
+
+        @Override
+        public URI serverUri(ParsedS3Request request, Optional<String> tableName, Optional<String> lowercaseAction, URI baseUri)
+        {
+            return UriBuilder.fromUri(baseUri).port(containerPort).build();
+        }
+
+        @Override
+        public Map<String, Object> toDocument(ParsedS3Request request, Optional<String> tableName, Optional<String> lowercaseAction)
+        {
+            return ImmutableMap.of("table", tableName);
+        }
+    }
+
+    private static final String POLICY = """
+            package test
+
+            default allow = false
+
+            allow {
+            	input.table = "good"
+            }
+            """;
+
+    @Inject
+    public TestOpaSecurity(@ForOpa HttpClient httpClient, OpaContainer container, S3Client s3Client)
+    {
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
+        containerPort = container.getPort();
+        this.s3Client = requireNonNull(s3Client, "s3Client is null");
+    }
+
+    @BeforeAll
+    public void setup()
+    {
+        Request request = preparePut()
+                .setUri(URI.create("http://localhost:%s/v1/policies/test".formatted(containerPort)))
+                .setHeader("Content-Type", "application/json")
```

Comment:
I believe this should be `text/plain` https://www.openpolicyagent.org/docs/latest/rest-api/#create-or-update-a-policy

The policy itself is rego, not json